CONTRIBUTING.md

```diff
@@ -19,53 +19,91 @@ Read the following guide if you're interested in contributing to cluster-api.
+If you're new to the project and want to help, but don't know where to start, we have a semi-curated list of issues that
+should not need deep knowledge of the system. [Have a look and see if anything sounds
+interesting](https://github.com/kubernetes-sigs/cluster-api/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22).
+Alternatively, read some of the docs on other controllers and try to write your own, file and fix any/all issues that
+come up, including gaps in documentation!
 
 ## Contributing a Patch
 
 1. If you haven't already done so, sign a Contributor License Agreement (see details above).
 1. Fork the desired repo, develop and test your code changes.
 1. Submit a pull request.
-  1. All code PR must be labeled with
-    ⚠️ (:warning:, major or breaking changes), ✨ (:sparkles:, minor or feature additions), 🐛 (:bug:, patch and bugfixes), 📖 (:book:, documentation or proposals), or 🏃 (:running:, other)
+    1. All code PR must be labeled with one of
+        - ⚠️ (:warning:, major or breaking changes)
+        - ✨ (:sparkles:, minor or feature additions)
+        - 🐛 (:bug:, patch and bugfixes)
+        - 📖 (:book:, documentation or proposals)
+        - 🏃 (:running:, other)
 
-All changes must be code reviewed. Coding conventions and standards are explained in the official [developer docs](https://github.com/kubernetes/community/tree/master/contributors/devel). Expect reviewers to request that you avoid common [go style mistakes](https://github.com/golang/go/wiki/CodeReviewComments) in your PRs.
+All changes must be code reviewed. Coding conventions and standards are explained in the official [developer
+docs](https://github.com/kubernetes/community/tree/master/contributors/devel). Expect reviewers to request that you
+avoid common [go style mistakes](https://github.com/golang/go/wiki/CodeReviewComments) in your PRs.
 
 ## Backporting a Patch
 
-Cluster API ships older versions through `release-X.X` branches, usually backports are reserved to critical bug-fixes.
-Some release branches might ship with both Go modules and dep (e.g. `release-0.1`), users backporting patches should always make sure
-that the vendored Go modules dependencies match the Gopkg.lock and Gopkg.toml ones by running `dep ensure`
+Cluster API maintains older versions through `release-X.Y` branches. We accept backports of bug fixes to the most recent
+release branch. For example, if the most recent branch is `release-0.2`, and the `master` branch is under active
+development for v0.3.0, a bug fix that merged to `master` that also affects `v0.2.x` may be considered for backporting
+to `release-0.2`. We generally do not accept PRs against older release branches.
+
+## Breaking Changes
+
+Breaking changes are generally allowed in the `master` branch, as this is the branch used to develop the next minor
```

Comment:
We should specify that this applies only if master is open for the next iteration

Reply:
I know Kubernetes sends out updates throughout the release cycle informing about the state of the master branch. Maybe we should do something similar (defer to #768?)

Reply:
That sounds good to me, let's document that as well?

Reply:
Yes, but as part of 768

Reply:
Eek, I linked to the wrong thing. Nevermind about 768 😄. We had talked at the f2f about formalizing the release cycle, which is what I was trying to link to, but I don't think we ever created an issue for it (or I can't find it quickly).

Reply:
Updated PTAL